backbone/src/main/java/org/researchstack/backbone/DataProvider.java

```diff
@@ -78,13 +80,25 @@ public static void init(DataProvider instance) {
     /**
      * Called to sign the user in to the backend service
      *
+     * @param context  android context
      * @param username the user's username
      * @param password the user's password
-     * @param context android context
      * @return Observable of the result of the method, with {@link DataResponse#isSuccess()}
      * returning true if signIn was successful
      */
-    public abstract Observable<DataResponse> signIn(Context context, String username, String password);
+    public abstract Observable<DataResponse> signIn(Context context, String username, String
+            password);
+
+    /**
+     * Request for a link to sign in user
+     *
+     * @param username the user's username
+     * @return Observable of the result of the method, with {@link DataResponse#isSuccess()}
+     * returning true if signIn was successful
+     */
+    public abstract Observable<DataResponse> requestSignInLink(String username);
```

Comment:
Having to maintain an abstract DataProvider is difficult.  Does it make sense to enforce the requestSignInLink(String username) for all devs?